manifests/init.pp

```diff
@@ -348,5 +353,13 @@
       Class['jenkins::firewall'] ->
         Anchor['jenkins::end']
   }
+
+  if $config_template {
+    Anchor['jenkins::begin'] ->
+      Class[$jenkins_package_class] ->
+        Class['jenkins::config'] ->
```

Comment:
There is a pre-existing ordering for between the package class and `jenkins::config` so this isn't necessary.  If you feel both of those class must come before it would be better to declare as coming before  `jenkins::config::global`.  The goal is to limit the code block to dealing with only the immediate concern.